lib/utils/mailup.ts

```diff
@@ -0,0 +1,287 @@
+/**
+ * Implements a Nodemailer MailUp transport.
+ *
+ * Uses the MailUp REST API to send transactional emails:
+ * see http://help.mailup.com/display/mailupapi/Transactional+Emails+using+APIs
+ *
+ */
+import * as t from "io-ts";
+import * as request from "superagent";
+
+import { Either, isLeft, isRight, left, right } from "fp-ts/lib/Either";
+import { EmailString, NonEmptyString } from "./strings";
+
+import * as nodemailer from "nodemailer";
+
+import { Address as NodemailerAddress } from "nodemailer/lib/addressparser";
+
+import * as winston from "winston";
+
+import { fromNullable, none, Option, some } from "fp-ts/lib/Option";
+import { ReadableReporter } from "./validation_reporters";
+
+// request timeout in milliseconds
+const DEFAULT_REQUEST_TIMEOUT_MS = 10000;
+
+// number of retries in case of HTTP errors
+const DEFAULT_REQUEST_RETRIES = 2;
+
+export const ENDPOINTS = {
+  sendTransactionalMail: "https://send.mailup.com/API/v2.0/messages/sendmessage"
```

Comment:
reason why this is not a constant by itself?

Reply:
because there are several endpoints in mailup api (ie. the endpoint to configure a new permitted sender address)

Reply:
so?

``` js
const SEND_TRANSACTIONAL_MAIL_ENDPOINT = "...";
const XYZ_ENDPOINT = "..."; 
```

or even better:

``` js
const SEND_TRANSACTIONAL_MAIL_ENDPOINT = process.env.MAILUP_SEND_TRANSACTIONAL_MAIL_ENDPOINT || "...";
// ...
```